services/cd-service/pkg/repository/transformer.go

```diff
@@ -686,6 +686,12 @@ func (c *CreateApplicationVersion) Transform(
 		if err != nil {
 			return "", GetCreateReleaseGeneralFailure(err)
 		}
+		err = state.checkUserTeamPermissions(ctx, transaction, state, c.Application, auth.PermissionCreateRelease, c.RBACConfig)
```

Comment:
is it necessary that `checkUserTeamPermissions` is not part of `checkUserPermissions`?
Or the other way around:
Could you create a wrapper function that does both, so that in the transformers we only need to call one?